src/Hydrator/ModelHydrator.php

```diff
@@ -13,30 +13,23 @@
  */
 final class ModelHydrator implements Hydrator
 {
+    private $arrayHydrator;
+    private $creatableFromArray;
+
+    public function __construct(ArrayHydrator $arrayHydrator, CreatableFromArray $creatableFromArray)
```

Comment:
Createable from array is a Model. 
Do you propose one Hydrator instance per model?